src/schema/v2/HeroUnit/heroUnitsConnection.ts

```diff
@@ -1,40 +1,59 @@
 import { connectionWithCursorInfo } from "../fields/pagination"
 import { HeroUnitType } from "./HeroUnitType"
-import { GraphQLString, GraphQLFieldConfig } from "graphql"
+import { GraphQLBoolean, GraphQLString, GraphQLFieldConfig } from "graphql"
 import { ResolverContext } from "types/graphql"
 import { pageable } from "relay-cursor-paging"
 import { convertConnectionArgsToGravityArgs } from "lib/helpers"
 import { paginationResolver } from "schema/v2/fields/pagination"
 
+export const tokenRequiredMesage =
+  "You need to pass a X-Access-Token header to perform this action"
+
+export const invalidArgsMessage =
+  "Invalid arguments - can't have both showAll and term"
+
+export const pickLoader = (args, context) => {
+  const { showAll, term } = args
+
+  if (showAll && term) throw new Error(invalidArgsMessage)
+
+  const {
+    authenticatedHeroUnitsLoader,
+    heroUnitsLoader,
+    matchHeroUnitsLoader,
+  } = context
+
+  if (term && !matchHeroUnitsLoader) throw new Error(tokenRequiredMesage)
+
+  const loader =
+    (term && matchHeroUnitsLoader) ||
+    (showAll && authenticatedHeroUnitsLoader) ||
+    heroUnitsLoader
+
+  return loader
+}
+
 const HeroUnitConnection = connectionWithCursorInfo({
   nodeType: HeroUnitType,
 })
 
 export const heroUnitsConnection: GraphQLFieldConfig<void, ResolverContext> = {
   type: HeroUnitConnection.connectionType,
   args: pageable({
+    showAll: {
+      type: GraphQLBoolean,
+      description: "If true will all hero units.",
+      defaultValue: false,
+    },
```

Comment:
I think a better argument for this would be `private`. Like why, as a consumer of this field, would I not want to `showAll`?

Reply:
Ok happy to follow the precedent!

Reply:
Hrm - can you link me to a place where we do this?

```
[tsserver] Identifier expected. 'private' is a reserved word in strict mode. Modules are automatically in strict mode.
```

Reply:
https://github.com/artsy/metaphysics/blob/cb99c685c09c1aa155419e75d548364d4629e480/src/schema/v2/me/savedArtworks.ts#L36

https://github.com/artsy/metaphysics/blob/cb99c685c09c1aa155419e75d548364d4629e480/src/schema/v2/partner/partnerArtworks.ts#L23

Reply:
ah ha so just never destructure it and instead call args.private - got it 👍